packages/block-library/src/term-name/edit.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	useBlockProps,
+	BlockControls,
+	AlignmentControl,
+	InspectorControls,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import {
+	ToggleControl,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { useToolsPanelDropdownMenuProps } from '../utils/hooks';
+import { unlock } from '../lock-unlock';
+
+const { HTMLElementControl } = unlock( blockEditorPrivateApis );
+
+export default function TermNameEdit( {
+	attributes,
+	setAttributes,
+	context: { termId, taxonomy },
+} ) {
+	const { textAlign, tagName = 'div', isLink } = attributes;
+
+	const term = useSelect(
+		( select ) => {
+			if ( ! termId || ! taxonomy ) {
+				return null;
+			}
+			return select( coreStore ).getEntityRecord(
+				'taxonomy',
+				taxonomy,
+				termId
+			);
+		},
+		[ termId, taxonomy ]
+	);
+
+	const termName = term?.name || __( 'Term Name' );
+
+	const blockProps = useBlockProps( {
+		className: clsx( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+
+	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
+
+	const TagName = tagName;
+
+	let termNameDisplay = termName;
+	if ( isLink ) {
+		const termLink = term?.link || '#';
+		termNameDisplay = (
+			<a href={ termLink } onClick={ ( e ) => e.preventDefault() }>
+				{ termName }
+			</a>
+		);
+	}
+
+	return (
+		<>
+			<BlockControls group="block">
+				<AlignmentControl
+					value={ textAlign }
+					onChange={ ( nextAlign ) => {
+						setAttributes( { textAlign: nextAlign } );
+					} }
+				/>
+			</BlockControls>
+			<InspectorControls group="advanced">
+				<HTMLElementControl
```

Comment:
Sorry, I was telling the wrong component 😅

To be precise, it's `HeadingLevelDropdown`. This component will be rendered in the toolbar and should also support the paragraph element.

https://github.com/WordPress/gutenberg/blob/5992373408d388c6cd9a190b495bd5318cbfdbbb/packages/block-editor/src/components/block-heading-level-dropdown/index.js#L38

However, it does not support the `div` element. Perhaps the default element for a block is the `p` element. What do you think?

Reply:
Thank you! This is much better. I've switched to `HeadingLevelDropdown` in https://github.com/WordPress/gutenberg/pull/72129/commits/13190801fab4b7b97e79a480662c11ea9ba2a19e.